samples/verona/lang.cc

```diff
@@ -1337,9 +1479,162 @@ namespace verona
       }};
   }
 
+  PassDef partialapp()
+  {
+    // This should happen after `lambda` (so that anonymous types get partial
+    // application), after `autocreate` (so that constructors get partial
+    // application), and after `defaultargs` (so that default arguments don't
+    // get partial application).
+
+    // This means that partial application can't be written in terms of lambdas,
+    // but instead has to be anonymous classes. There's no need to check for
+    // non-local returns.
+    return {
+      dir::bottomup | dir::once,
+      {T(Function)[Function]
+         << ((T(Ref) / T(DontCare))[Ref] * Name[Id] *
+             T(TypeParams)[TypeParams] * T(Params)[Params]) >>
+       [](Match& _) {
+         auto f = _(Function);
+         auto ref = _(Ref);
+         auto id = _(Id);
+         auto tp = _(TypeParams);
+         auto params = _(Params);
+         size_t start_arity = 0;
+         auto end_arity = params->size();
+         auto parent = f->parent()->parent();
+         auto defs = parent->lookdown(id->location());
+         auto tn = parent->at(wf / Class / Ident, wf / TypeTrait / Ident);
+         Node call = (ref->type() == Ref) ? CallLHS : Call;
+
+         for (auto def : defs)
+         {
+           if ((def == f) || (def->type() != Function))
+             continue;
+
+           auto arity = def->at(wf / Function / Params)->size();
+
+           if ((arity >= start_arity) && (arity < end_arity))
+             start_arity = arity + 1;
+         }
+
+         Nodes names;
+
+         // Create a unique anonymous class name for each arity.
+         for (auto arity = start_arity; arity < end_arity; ++arity)
+           names.push_back(Ident ^ _.fresh());
+
+         Node ret = Seq;
+
+         for (auto arity = start_arity; arity < end_arity; ++arity)
+         {
+           // Create an anonymous class for each arity.
+           auto idx = arity - start_arity;
```

Comment:
```suggestion
           auto name = names[arity - start_arity];
```
and then replace `names[idx]` with `name`?